scala-package/macros/src/main/scala/org/apache/mxnet/APIDocGenerator.scala

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.mxnet
+
+import org.apache.mxnet.init.Base._
+
+import scala.collection.mutable.ListBuffer
+
+private[mxnet] object APIDocGenerator{
+  case class traitArg(argName : String, argType : String, argDesc : String, isOptional : Boolean)
+  case class traitFunction(name : String, desc : String,
+                           listOfArgs: List[traitArg], returnType : String)
+
+
+  def main(args: Array[String]) : Unit = {
+    val FILE_PATH = args(0)
+    absClassGen(FILE_PATH, true)
+    absClassGen(FILE_PATH, false)
+  }
+
+  def absClassGen(FILE_PATH : String, isSymbol : Boolean) : Unit = {
+    // scalastyle:off
+    val traitFunctions = initSymbolModule(isSymbol)
+    val traitfuncs = traitFunctions.filterNot(_.name.startsWith("_")).map(traitfunction => {
```

Comment:
can we have these filtering condition in one place, may be declare a constant string somewhere...its in too many places across files and easy to miss if we refactor

Reply:
Let's put them all together in one class, will create a Class for that